source/speechDictHandler/__init__.py

```diff
@@ -115,9 +114,22 @@ def sub(self, text):
 				del self[index]
 		return text
 
+
+# useful for #14689 (see below)
+IGNORE_TRAILING_WHITESPACE_LENGTH = 100
+
 def processText(text):
 	if not globalVars.speechDictionaryProcessing:
 		return text
+	# #14689: API level < 2 UIA consoles have many blank lines, which slows processing to a halt
+	# Because from UIAHandler.constants import WinConsoleAPILevel
+	# cause crashes as global import, and probably slowdowns here
+	# use 2 instead of WinConsoleAPILevel.FORMATTED constant
+	focus = api.getFocusObject()
+	if hasattr(focus, "apiLevel") and focus.apiLevel < 2:

```

Comment:
I'd really like a more precise check than `hasattr`, especially as NVDA can run external code in add-ons. Is there really no way of using `isinstance`?